app/src/main/java/com/codesoom/assignment/TaskHttpHandler.java

```diff
@@ -0,0 +1,241 @@
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+/**
+ *  Task의 로직을 가지고 있고 관련된 Http 요청을 처리하는 클래스
+ */
+public class TaskHttpHandler implements HttpHandler {
+    private final List<Task> tasks = new ArrayList<>();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private Long id = 0L;
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        String method = exchange.getRequestMethod();
+        URI uri = exchange.getRequestURI();
+        String path = uri.getPath();
+
+        String content = "";
+        String request = parsingRequest(exchange.getRequestBody());
+
+        if (method.equals("GET")) {
+            if (isDetailMatches(path)) {
+                try {
+                    content = handleDetailGet(path);
+                    exchange.sendResponseHeaders(200, content.getBytes().length);
+                } catch (NoSuchElementException e) {
+                    exchange.sendResponseHeaders(404, -1);
+                }
+            } else {
+                content = tasksToJson();
+                exchange.sendResponseHeaders(200, content.getBytes().length);
+            }
+        } else if (method.equals("POST") && path.equals("/tasks")) {
+            content = handlePost(request);
+            exchange.sendResponseHeaders(201, content.getBytes().length);
+        } else if (method.equals("PUT") && isDetailMatches(path)) {
+            try {
+                content = handlePut(path, request);
+                exchange.sendResponseHeaders(200, content.getBytes().length);
+            } catch (NoSuchElementException e) {
+                exchange.sendResponseHeaders(404, -1);
+            }
+        } else if (method.equals("DELETE") && isDetailMatches(path)) {
+            try {
+                content = handleDelete(path);
+                exchange.sendResponseHeaders(204, -1);
+            } catch (NoSuchElementException e) {
+                exchange.sendResponseHeaders(404, -1);
+            }
+        } else {
+            content = handleBadRequest();
+            exchange.sendResponseHeaders(400, content.getBytes().length);
+        }
+
+        OutputStream responseBody = exchange.getResponseBody();
+        responseBody.write(content.getBytes());
+        responseBody.flush();
+        responseBody.close();
+    }
+
+    /**
+     * Task의 id가 경로에 포함되어 있으면 true, 아니면 false를 리턴합니다.
+     * @param path 수신된 Http 요청의 경로
+     * @return id가 경로에 포함되어 있으면 true, 아니면 fasle
+     */
+    private boolean isDetailMatches(String path) {
+        return path.matches("/tasks/[0-9]+");
+    }
+
+    /**
+     * 특정 Task를 GET 요청이 왔을 때, 요청한 Task를 리턴한다.
+     * @param path 수신된 Http 요청의 경로
+     * @return 요청한 Task를 String으로 변환해서 리턴한다.
+     * @throws JsonProcessingException Json으로 변환하면서 에러가 발생한 경우 던집니다.
+     * @throws NoSuchElementException 요청된 id를 찾지 못했을 경우 던집니다.
```

Comment:
```suggestion
     * 특정 Task를 GET 요청이 왔을 때, 요청한 Task를 리턴한다.
     * 
     * @param path 수신된 Http 요청의 경로
     * @return 요청한 Task를 String으로 변환해서 리턴한다.
     * @throws JsonProcessingException Json으로 변환하면서 에러가 발생한 경우 던집니다.
     * @throws NoSuchElementException 요청된 id를 찾지 못했을 경우 던집니다.
```
nit : 보통 javadoc 을 작성할 때 어노테이션이 나열되는 부분과 설명 사이에는 빈 한줄을 추가합니다.
아래 오라클 링크의 예시가 도움이 될 것 같아요!
https://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html

Reply:
이런 컨벤션이 있었군요. 감사합니다.

Reply:
@ssisksl77 과제와 상관은 없지만 질문이 있습니다.
참고할 javadoc를 java8 docs에서 주셨는데 java18 docs에도 javadoc 가이드는 있습니다. 8을 보내주신 이유가 따로 있으신가요?
저는 공식문서를 볼 때 어느 버전의 doc를 참고하는게(찾는 정보가 두 버전에 모두 있을시에) 더 나을지 항상 혼란스러워 질문드립니다.

Reply:
오! 두 버전에 큰 차이가 없을텐데 어느부분이 혼란스러우셨나요?

Reply:
저는 일반적으로 공식문서를 볼 때 제가 사용하는 버전의 javadoc을 보게되는 것 같아요!